### PR TITLE
Issue 2.1-Implement JDBC connection (Driver for class)

### PR DESCRIPTION
**Pull request by Skywalker303**

### CODE REVIEW

**Comment by MrToady** [comment]
Please move @Bonoparte from reviewers and add @acorned

File: build.gradle [hunk 1]
```diff
@@ -14,7 +14,8 @@ dependencies {
             'org.springframework:spring-core:5.0.2.RELEASE',
             'org.apache.tomcat:tomcat-catalina:9.0.2',
             'org.apache.tomcat:tomcat-servlet-api:9.0.2'
-
+    compile group: 'c3p0', name: 'c3p0', version: '0.9.0.4'
```
Comment (MrToady) [inline build.gradle]: use gradle style.

File: src/main/resources/jdbc/jdbc.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+driver=org.postgresql.Driver
```
Comment (MrToady) [inline src/main/resources/jdbc/jdbc.properties:1]: I like your decision to create a property file!

File: src/main/java/DbConnection/DataSourceInit.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package DbConnection;
+
+import com.mchange.v2.c3p0.ComboPooledDataSource;
+import temp.Market;
+
+import java.beans.PropertyVetoException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
```
Comment (MrToady) [inline src/main/java/DbConnection/DataSourceInit.java:10]: redundant blank line)

File: src/main/java/DbConnection/JdbcConnector.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        try (InputStream resourceAsStream = JdbcConnector.class.getResourceAsStream("resources\\jdbc.properties")) {
+            properties.load(resourceAsStream);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            Class.forName(properties.getProperty("driver"), true, ComboPooledDataSource.class.getClassLoader());
+        } catch (Exception e) {
+            System.out.println(" error while instantiating JdbcDriver");
```
Comment (MrToady) [inline src/main/java/DbConnection/JdbcConnector.java]: better use printStackTrace

File: src/main/java/DbConnection/JdbcConnector.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import com.mchange.v2.c3p0.ComboPooledDataSource;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+
+public class JdbcConnector {
+    final Properties properties = new Properties();
+
+    private void getcon() throws IOException {
```
Comment (MrToady) [inline src/main/java/DbConnection/JdbcConnector.java]: We haven't any constraints about variable length. We can name this method "getConnection()" for example.

File: src/main/java/DbConnection/DataSourceInit.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+public final class DataSourceInit {
+    private static ComboPooledDataSource instance;
+
+    public static ComboPooledDataSource getDataSource() {
```
Comment (MrToady) [inline src/main/java/DbConnection/DataSourceInit.java]: Use interfaces here. You don't need to return concrete class. Other code shouldn't know about your implementation.

**Review by acorned — APPROVED** [review]

**Comment by Skywalker303** [comment]
All comments are fixed

**Pull request merged**